test/functional/wallet_ancient_migration.py

```diff
@@ -0,0 +1,292 @@
+#!/usr/bin/env python3
+# Copyright (c) 2025-present The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""
+Test wallet migration from ancient versions to current version.
+Tests two wallet types that are actually testable:
+1. Non-HD wallet from v0.14.3
+2. HD wallet from v0.14.3 (VERSION_HD_BASE)
+"""
+import os
+import shutil
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import (
+    assert_equal,
+    get_datadir_path,
+    initialize_datadir,
+)
+
+class WalletMigrationTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.setup_clean_chain = True
+        self.num_nodes = 2
+
+    def skip_test_if_missing_module(self):
+        self.skip_if_no_wallet()  # Skip if wallet is not compiled in
+        self.skip_if_no_previous_releases()
+
+    def setup_network(self):
+        # Don't setup nodes here, we'll do it per test
+        self.nodes = []
```

Comment:
nit: I think if you call `cleanup_nodes()` from within `setup_network()`, you don't need to call it from `migrate_wallet()`...
```suggestion
        self.nodes = []
        cleanup_nodes()
```